### ISSUE TITLE
[Animation]const expression with spread operator not working on production build

### ISSUE THREAD

**Issue by bob-lee** [angular/angular#25317]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior
Have 2 `const` variables (taken from [matsko/ng4-animations-preview](https://github.com/matsko/ng4-animations-preview/blob/master/src/app/image-modal/image-modal.component.ts#L11)) to be used in `style` function as below:
```
const zoomFadeIn = { opacity: 0, transform: 'translateX({{ x }}) translateY({{ y }}) scale(0)' };
const zoomFadeInFrom = { ...zoomFadeIn, transformOrigin: '{{ ox }} {{ oy }}' };
```
During development, they seem to work ok. 
Problem is that second const `zoomFadeInFrom` doesn't work silently on production build.
Spread operator used in object literal seems to the reason, I had to replace it with below code to make it work:
```
// query('.container', [style(zoomFadeInFrom)]), // not working silently
query('.container', [style({ opacity: 0, transform: 'translateX({{ x }}) translateY({{ y }}) scale(0)', transformOrigin: '{{ ox }} {{ oy }}' })]),
```

## Minimal reproduction of the problem with instructions
https://stackblitz.com/edit/angular-r6mynh?file=src%2Fapp%2Fform-modal.component.ts

This demo doesn't actually show the problem as Stackblitz seems not using AOT build..

## Environment

<pre><code>
Angular version: 6.0.3

Browser:
- [x ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 8.11.2
- Platform:  Windows 7 64-bit

**Comment by median-man** [comment]
I just ran into the same problem.

**Comment by adamduren** [comment]
I filed a bug report under `angular-cli` and found this after the fact. I believe to be having the same issue. https://github.com/angular/angular-cli/issues/13833 contains the infromation as well as a repository that has a minimal reproduction of the issue.

**Comment by adamduren** [comment]
@bob-lee Did you ever fix this? See my comment https://github.com/angular/angular-cli/issues/13833#issuecomment-470184395. Adding `export` to any `const` involved in the animation fixed the issue for me.

**Comment by sprinter-com-pl** [comment]
I have encountered the same problem today - my `const` object filled with properties from other `const` object using spread operator was broken on production build. My code was:

```
const AuthGuards = {
    canActivate: [RouteGuard, NgxPermissionsGuard],
    runGuardsAndResolvers: 'always'
}

export const AppRoutes = [    
    {
        path: '',
        component: DashboardComponent,
        ...AuthGuards
    },
   (a bunch of another objects with ...AuthGuards line)
]
```

Unfortunately, adding `export` to `AuthGuards` doesn't work for me. I had to fix it by removing `...AuthGuards` line and repeating  `AuthGuards` properties in every route object which requires them.

**Comment by xxGizmoCodexx** [comment]
Facing the same type of problem where spread operator is evaluated to `null` in prod builds

Short e.g.

#### Usage
```typescript
return transportConfig.provider.configure(transportConfig.providerKey, ...transportConfig.configArgs);
```

#### Evaluation
```typescript
  public static configure(providerKey: string, hubConfig: HubConfig): ModuleWithProviders {


    // function providerFact(signalr: SignalRService) {
    //   const signalRProvider = new SignalRTransportProvider(signalr, hubConfig);
    //   return {
    //     providerKey: providerKey,
    //     provider: signalRProvider
    //   };
    // }
};
```

**Comment by trotyl** [comment]
Trivial note: there is no such thing as "spread operator", spread is not an operator.

+ https://github.com/kangax/compat-table/pull/1331
+ https://github.com/tc39/ecma262/pull/1557

**Comment by twisha** [comment]
> Trivial note: there is no such thing as "spread operator", spread is not an operator.
> 
> * [kangax/compat-table#1331](https://github.com/kangax/compat-table/pull/1331)
> * [tc39/ecma262#1557](https://github.com/tc39/ecma262/pull/1557)

spread operator is an operator. Refer https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax

Check the ngrx documentation as well - https://ngrx.io/guide/store/reducers

**Comment by JoostK** [comment]
This has been fixed with Ivy in Angular 9.0!
